config/config.go

```diff
@@ -158,36 +173,51 @@ func GetScriptDirectory() string {
 	return cachedScriptDir
 }
 
+//SetConfigFile sets the configDirectory cache to the entered value,
+//bypassing how cordless sets defaults.
+func SetConfigDirectory(configPath string) (string, error) {
+	checkConfig := CheckConfigDirectory(configPath)
+	if checkConfig != nil {
+		return "", checkConfig
+	}
+	cachedConfigDir = configPath
+	return cachedConfigDir, nil
+}
 //GetConfigDirectory is the parent directory in the os, that contains the
 //settings for the application.
 func GetConfigDirectory() (string, error) {
 	if cachedConfigDir != "" {
 		return cachedConfigDir, nil
 	}
 
-	directory, err := getConfigDirectory()
+	directory, err := getDefaultConfigDirectory()
 	if err != nil {
 		return "", err
 	}
 
-	_, statError := os.Stat(directory)
-	if os.IsNotExist(statError) {
-		//Folders have to be executable for some reason, therefore 766 instead of 666.
-		createDirsError := os.MkdirAll(directory, 0766)
-		if createDirsError != nil {
-			return "", createDirsError
-		}
-	} else if statError != nil {
-		return "", statError
+	checkConfig := CheckConfigDirectory(directory)
+	if checkConfig != nil {
+		return "", checkConfig
 	}
-
 	//After first retrieval, we will save this, as we needn't redo all that
 	//stuff over and over again.
 	cachedConfigDir = directory
 
 	return cachedConfigDir, nil
 }
 
+func CheckConfigDirectory(directoryPath string) error {
```

Comment:
If this method isn't called from outside the `config` package, it can start with a lowercase letter. Golang uses the casing in order to define visibility. Uppercase means public and lowercase private.

Reply:
Thank you, I forgot that in GO function names actually matter